apps/readest-app/src/app/reader/components/ReaderContent.tsx

```diff
@@ -44,9 +46,22 @@ const ReaderContent: React.FC<{ ids?: string; settings: SystemSettings }> = ({ i
   const [showDetailsBook, setShowDetailsBook] = useState<Book | null>(null);
   const isInitiating = useRef(false);
   const [loading, setLoading] = useState(false);
+  const [activeBookKey, setActiveBookKey] = useState<string | null>(null);
 
   useBookShortcuts({ sideBarBookKey, bookKeys });
 
+  useEffect(() => {
+    setActiveBookKey(sideBarBookKey || (bookKeys.length > 0 ? bookKeys[0]! : null));
+  }, [sideBarBookKey, bookKeys]);
+
+  const {
+    syncState,
+    conflictDetails,
+    resolveConflictWithLocal,
+    resolveConflictWithRemote,
+    pushProgress,
+  } = useKOSync(activeBookKey || '');
```

Comment:
It's better to use the `useKOSync` hook in each `FoliateViewer` just as `useProgressSync` hook does. In this way multiple documents (in Parallel Read) could be synced in parallel.